core/forms.py

```diff
@@ -107,6 +107,31 @@ def save(self, commit=True):
             self.save_m2m()
         return instance
 
+    @property
+    def hydrated_fielsets(self):
+        # for some reason self.fields returns defintions and not bound fields
+        # so until i figure out a better way we can just create a dict here
+        # https://github.com/django/django/blob/main/django/forms/forms.py#L52
+
+        bound_field_dict = {}
+        for field in self:
+            bound_field_dict[field.name] = field
+
+        hydrated_fieldsets = []
+
+        for fieldset in self.fieldsets:
+            hyrdrated_fieldset = {
+                "layout": fieldset["layout"],
+                "layout_attrs": fieldset.get("layout_attrs", {}),
+                "fields": [],
+            }
+            for field_name in fieldset["fields"]:
+                hyrdrated_fieldset["fields"].append(bound_field_dict[field_name])
+
+            hydrated_fieldsets.append(hyrdrated_fieldset)
+
+        return hydrated_fieldsets
+
```

Comment:
Hm yeah this seems... hacky. But I'm also unsure what other approach would work here. Regardless lets implement on the other models as well. I'm good with the idea generally.

Reply:
My issue with the current api is that It will be confusing for a new person 

In order to add a field, you must 

add it to fields in meta, then add it to field sets only then will it show

if something is added to fieldsets and not to fields then it will do nothing.

i think ideally the api should be

```
class SleepForm(CoreModelForm, TaggableModelForm):


    class Meta:
        model = models.Sleep
        fieldsets = [
            {
                "fields": ["child", "start", "end", "nap"],
                "layout": "required",
            },
            {"layout": "advanced", "fields": ["notes", "tags"]},
        ]
        widgets = {
            "child": ChildRadioSelect,
            "start": DateTimeInput(),
            "end": DateTimeInput(),
            "notes": forms.Textarea(attrs={"rows": 5}),
        }
```

Reply:
gave it a try but will require much deeper work to do it properly

<img width="1202" alt="image" src="https://github.com/babybuddy/babybuddy/assets/1640136/423c8d8f-e85b-4834-bf82-7170ba65f42e">